### PR TITLE
fix(38463): No indexed access completions after Record<...>

### LINKED ISSUE #38463: No indexed access completions after Record<...>
```ts
interface Foo {
    hello: string;
    world: boolean;
}

type Yadda = Record<keyof Foo, any>["/**/"];
```

Request completions at `/**/`.

**Expected**: `foo`, `bar`
**Actual**: No completions

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #38463

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@typescript-bot pack this

**Comment by DanielRosenwasser** [comment]
Does this also fix #38667?

**Comment by a-tarasyuk** [comment]
> Does this also fix #38667?

It seems, yes.

Screenshot 2020-05-20 at 09 50 54

The problem is filtering private identifiers because to check if a property is a private identifier, _completions_ use`.valueDeclaration`, which in some cases may be undefined. I'm not sure that even we need to use `prop.syntheticOrigin.valueDeclaration`, maybe just checking for .valueDeclaration` is enough. What do you think?

File: src/services/stringCompletions.ts [hunk 1]
```diff
@@ -223,13 +223,16 @@ namespace ts.Completions.StringCompletions {
     function stringLiteralCompletionsFromProperties(type: Type | undefined): StringLiteralCompletionsFromProperties | undefined {
         return type && {
             kind: StringLiteralCompletionKind.Properties,
-            symbols: type.getApparentProperties().filter(prop =>
-                !isPrivateIdentifierPropertyDeclaration(
-                    isTransientSymbol(prop) && prop.syntheticOrigin ? prop.syntheticOrigin.valueDeclaration : prop.valueDeclaration)),
+            symbols: type.getApparentProperties().filter(isPropertyNotPrivate),
```
Comment (DanielRosenwasser) [inline src/services/stringCompletions.ts:226]: While you're here, can you switch to the built-in `filter` in `core.ts` which avoids an allocation if everything is the same.

**Comment by DanielRosenwasser** [comment]
I think `valueDeclaration` should be enough - I'd give it a shot.

> It seems, yes.

Would be super grateful if you could add it as a test. 🙂

File: src/services/stringCompletions.ts (hunk 1 above)
Comment (a-tarasyuk) [reply src/services/stringCompletions.ts:226]: Sure

**Comment by a-tarasyuk** [comment]
Ok, I'll update PR in 15 minutes.

**Comment by a-tarasyuk** [comment]
> I think valueDeclaration should be enough - I'd give it a shot.
> Would be super grateful if you could add it as a test. 

@DanielRosenwasser Fixed :)

**Pull request merged**